solc/CommandLineParser.cpp

```diff
@@ -417,36 +419,10 @@ void CommandLineParser::parseLibraryOption(string const& _input)
 					"Note that there should not be any whitespace after the " +
 					(isSeparatorEqualSign ? "equal sign" : "colon") + "."
 				);
-
-			if (addrString.substr(0, 2) == "0x")
-				addrString = addrString.substr(2);
-			else
-				solThrow(
-					CommandLineValidationError,
-					"The address " + addrString + " is not prefixed with \"0x\".\n"
-					"Note that the address must be prefixed with \"0x\"."
-				);
-
-			if (addrString.length() != 40)
-				solThrow(
-					CommandLineValidationError,
-					"Invalid length for address for library \"" + libName + "\": " +
-					to_string(addrString.length()) + " instead of 40 characters."
-				);
-			if (!util::passesAddressChecksum(addrString, false))
-				solThrow(
-					CommandLineValidationError,
-					"Invalid checksum on address for library \"" + libName + "\": " + addrString + "\n"
-					"The correct checksum is " + util::getChecksummedAddress(addrString)
-				);
-			bytes binAddr = util::fromHex(addrString);
-			util::h160 address(binAddr, util::h160::AlignRight);
-			if (binAddr.size() > 20 || address == util::h160())
```

Comment:
Oh actually the test for empty address was removed from the command-line so maybe it is fine indeed to also not have it in the Standard JSON, right @cameel?

Reply:
I missed that in my review. Although this is checked later, I think it is simple enough that the redundancy is harmless and worth the extra safety.

Reply:
@ matheusaaguiar, just to be on the safe side - I assume that I should make the changes according to this comments.  Is this correct?  I am asking to make sure that this is not an internal discussion.

Reply:
@matheusaaguiar, just to be on the safe side - I assume that I should make the changes according to this comments. Is this correct? I am asking to make sure that this is not an internal discussion.

Reply:
@matheusaaguiar since @davidBar-On is already checked the address length here: https://github.com/ethereum/solidity/pull/14283/files#diff-38ee70850074a54efd416d53701471fa79742fd73f95e390e020beb01fdac93bR41 and we are now using the same `validateAddress` function that he created for both command-line and standard JSON, I think it is not necessary to check for empty address. But yeah, I just pointed it out so we are all aware that the check for an empty address was removed.

@davidBar-On I don't think you need to do anything in this regard. But let's see what @cameel has to say, since he was the one that mentioned such a check.